packages/stark-ui/src/modules/session-ui/routes.ts

```diff
@@ -16,6 +18,20 @@ import {
 	StarkSessionLogoutPageComponent
 } from "./pages";
 
+/**
+ * Hook to destroy the OverlayContainer inside which all overlays are rendered (i.e. stark-dropdown options)
+ * Fixes https://github.com/NationalBankBelgium/stark/issues/1570
+ */
+export function destroyOverlaysOnEnterFn(transition: Transition): HookResult {
+	// inject the OverlayContainer
+	const overlayContainer: OverlayContainer = transition.injector().get<OverlayContainer>(OverlayContainer);
+	// destroy the container by calling its own "ngOnDestroy" method
+	// see https://github.com/angular/components/pull/5378/files
+	/* tslint:disable-next-line:no-lifecycle-call*/
+	overlayContainer.ngOnDestroy();
```

Comment:
What would happen if no instance of `OverlayContainer` is available because the application does not use it ?

I don't have a good example in mind but wouldn't it be better to put this in a `if` condition ? 😊 
```js
if (typeof overlayContainer !== "undefined") {
    overlayContainer.ngOnDestroy();
}
```

Reply:
Good point... indeed, that might happen if the app don't use any of the stark components that create an overlay (dropdown, app-data, menu, etc).

I'll add the check 😉

Reply:
Mmm... I have a small doubt now 😛 
Did you already put the try/catch before?

Because with the try/catch, I'm not sure the `if` is really necessary.
If it was there before, my bad I didn't see it.

Reply:
The `if` wasn't there before, but I added it together with the `try ... catch` just in case we get something injected  but `undefined`. I know it doesn't make not much sense... but I added it just in case... to be on the safe side...

Reply:
In the end I removed the `if` check cause it really doesn't make sense... the `try...catch` is more than enough ;)